### DISCUSSION TITLE
How to display the value from relationship in function form?

### DISCUSSION THREAD

**Discussion by jigujigu** [filamentphp/filament#3953] [Help] [answered]
Hi All,

How to display the value from relationship in function form?

It's showing in the list, but not in edit/view form.
![image](https://user-images.githubusercontent.com/6214964/189504507-53c4ac10-0ee7-4590-9ea5-d5ca9ed88c69.png)

![image](https://user-images.githubusercontent.com/6214964/189504520-37c18557-efc2-4d20-bf23-3bc7bd87fa47.png)

```
    public static function form(Form $form): Form
    {
        return $form
            ->columns(2)
            ->schema([
                \Filament\Forms\Components\TextInput::make('order_number')
                    ->label('#Order')

                \Filament\Forms\Components\DatePicker::make('order_date')
                    ->label('Date')

                \Filament\Forms\Components\Select::make('vendor_id')
                    ->label('Vendor')
                    ->relationship('vendor', 'name')
                    ->reactive()
                    ->afterStateUpdated(function ($state, callable $set) {
                        $vendor = Vendor::find($state);
                        if ($vendor) {
                            $set('vendor.address', $vendor->address);
                        }
                    }),

                \Filament\Forms\Components\TextArea::make('vendor.address')
                    ->label('Address')
                    ->disabled(),
            ]);
    }

    public static function table(Table $table): Table
    {
        return $table
            ->columns([
                Tables\Columns\TextColumn::make('order_number')->label('#Order'),

                Tables\Columns\TextColumn::make('vendor.name')->label('Vendor'),
                Tables\Columns\TextColumn::make('vendor.address')->label('address'),

            ])
            ->filters([
                //
            ])
            ->actions([
                Tables\Actions\ViewAction::make(),
                Tables\Actions\EditAction::make(),
            ])
            ->bulkActions([
                Tables\Actions\DeleteBulkAction::make(),
            ]);
    }
```

Thank you.

**Comment by jigujigu** [comment, marked as answer]
Hi All,

I use Closure $get, then the field automatic filled each time select component is changed.

Thank You

                \Filament\Forms\Components\Select::make('vendor_id')
                    ->label('Vendor')
                    ->relationship('vendor', 'name')
                    ->reactive()

                \Filament\Forms\Components\TextArea::make('vendor.address')
                    ->label('Address')
                    ->disabled(),
                    ->placeholder(
                        function (Closure $get) {
                            $vendor_id = $get('vendor_id');
                            $vendor = Vendor::find($vendor_id);
                             return ($vendor)? $vendor->address: "";
                        }
                    ),

``

**Reply by danharrin** [reply]
You actually need this - https://filamentphp.com/docs/2.x/forms/layout#saving-data-to-relationships

This loads AND saves the data to a relationship automatically
